File: 169-majority-element/majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        int n=nums.length;
        HashMap <Integer,Integer> map = new HashMap<>();
        for(int i:nums){
            map.put(i,map.getOrDefault(i,0)+1);
        }  
        for(int i:map.keySet()){
            if(map.get(i)>n/2){
                return i; 
            }
        } 
        return -1;
    }
}

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        # max_num = 0
        # n=nums[0]
        # for i in range(len(nums)):
        #     count = 0
        #     for j in range(i,len(nums)):
        #         if nums[i] == nums[j]:
        #                 count += 1
        #     if count > max_num:
        #         max_num = count 
        #         n = nums[i]
        # return n    

        freq = {}
        for i in range(len(nums)):
            freq[nums[i]] = freq.get(nums[i],0)+1
        count = 0    
        for i in freq:
            if freq[i] > count:
                count = freq[i]
                n = i
        return n   
                      



        